helpers/backfills.py

```diff
@@ -0,0 +1,70 @@
+import logging
+
+import shared.torngit as torngit
+from asgiref.sync import async_to_sync
+from sqlalchemy.orm.session import Session
+
+from database.models.core import GithubAppInstallation, Repository
+
+log = logging.getLogger(__name__)
+
+# GH App Backfills
+# Looping and adding all repositories in the installation app
+def add_repos_service_ids_from_provider(
+    db_session: Session,
+    ownerid: int,
+    owner_service: torngit.base.TorngitBaseAdapter,
+    gh_app_installation: GithubAppInstallation,
+):
+    repos = async_to_sync(owner_service.list_repos_using_installation)()
+
+    if repos:
+        # Fetching all repos service ids we have for that owner in the DB
+        repo_service_ids_in_db = [
+            repo.service_id
+            for repo in db_session.query(Repository.service_id)
+            .filter_by(ownerid=ownerid)
+            .all()
+        ]
+
+        # Add service ids from provider that we have DB records for to a list
+        new_repo_service_ids = []
+        for repo in repos:
+            repo_data = repo["repo"]
+            service_id = repo_data["service_id"]
+            if service_id and service_id in repo_service_ids_in_db:
+                new_repo_service_ids.append(service_id)
+        log.info(
+            "Added the following repo service ids to this gh app installation",
+            extra=dict(
+                ownerid=ownerid,
+                installation_id=gh_app_installation.installation_id,
+                new_repo_service_ids=new_repo_service_ids,
+            ),
+        )
+        gh_app_installation.repository_service_ids = new_repo_service_ids
```

Comment:
Won't this overwrite what might be already there? Or are you sure at this point that the `repository_service_ids` is an empty list (when it shouldn't be)?

Reply:
Yeah this won't overwrite the things we have, see here for example: https://github.com/codecov/worker/blob/163e7c9f933163ecfa4fa094863c4b5f68ddeadb/tasks/backfill_existing_gh_app_installations.py#L61-L62. We first check if the selection is all, else get independent repos for each app. Same happens for the other task, https://github.com/codecov/worker/blob/163e7c9f933163ecfa4fa094863c4b5f68ddeadb/tasks/backfill_owners_without_gh_app_installations.py#L83-L84

Reply:
What I meant is, for example, if some installation `G` has `repository_service_ids={1200}`. Then this logic would overwrite the current selection `{1200}`.

But it doesn't really matter cause you are listing _all_ the repos the installation has access to, which would include `1200`.